src/H5T.c

```diff
@@ -5660,8 +5660,8 @@ H5T__path_find_real(const H5T_t *src, const H5T_t *dst, const char *name, H5T_co
     old_npaths = H5T_g.npaths;
 
     /* Set a few convenience variables */
-    new_api_func = (matched_path && conv->is_app && conv->u.app_func);
```

Comment:
Add an additional check to prevent the no-op path from being considered a new conversion path, which had some odd side effects of replacing the path with a slightly different, new no-op path.